gnovm/pkg/doc/print.go

```diff
@@ -113,35 +109,29 @@ func (pkg *pkgPrinter) newlines(n int) {
 	}
 }
 
-// emit prints the node. If pkg.opt.Source is true, it ignores the provided comment,
+// emit prints the node signature. If pkg.opt.Source is true, it ignores the provided comment,
 // assuming the comment is in the node itself. Otherwise, the go/doc package
 // clears the stuff we don't want to print anyway. It's a bit of a magic trick.
-func (pkg *pkgPrinter) emit(comment string, node ast.Node) {
-	if node != nil {
-		var arg any = node
-		if pkg.opt.Source {
-			// Need an extra little dance to get internal comments to appear.
-			arg = &printer.CommentedNode{
-				Node:     node,
-				Comments: pkg.file.Comments,
-			}
-		}
-		err := format.Node(&pkg.buf, pkg.fs, arg)
-		if err != nil {
-			pkg.Fatalf("%v", err)
-		}
+func (pkg *pkgPrinter) emit(comment string, node string) {
+	if node != "" {
+		pkg.Printf("%s\n", node)
 		if comment != "" && !pkg.opt.Source {
 			pkg.newlines(1)
-			pkg.ToText(&pkg.buf, comment, indent, indent+indent)
+			ToText(&pkg.buf, comment, indent, indent+indent)
 			pkg.newlines(2) // Blank line after comment to separate from next item.
 		} else {
 			pkg.newlines(1)
 		}
 	}
 }
 
-// oneLineNode returns a one-line summary of the given input node.
-func (pkg *pkgPrinter) oneLineNode(node ast.Node) string {
+// oneLineType parses the expression and returns a one-line summary of the node.
+func (pkg *pkgPrinter) oneLineType(expression string) string {
+	node, err := parser.ParseExpr(expression)
```

Comment:
Best to use `ParseExprFrom` with a token.Fileset we re-use later: this ensures that comments stay in the same positions.

Reply:
To make a `token.FileSet`, we need the original source file, right? But `oneLineType` is called with the `Type` string from a `JSONField` which doesn't have the original source file. What should be used to build the `token.FileSet`?

Reply:
You say that `ParseExprFrom` ensures that comments stay in the same positions. But the purpose of `oneLineType` is to produce a simplified representation that strips out the comments. The inside of a struct, including comments, is replaced with `struct{ ... }`. I try to create a type with comments, like this:
```
type FuncType func(
	x int, // first param
) int
``` 
Even the original `gno doc` replaces this with `type FuncType func(x int) int`. So it seems that we don't need to preserve the position of comments in `oneLineType`, since there are no comments.